doc/porting/cmsis_rtos_v2.rst

```diff
@@ -0,0 +1,63 @@
+.. _cmsis_rtos_v2:
+
+CMSIS RTOS v2
+##########################
+
+Cortex-M Software Interface Standard (CMSIS) RTOS is a vendor-independent
+hardware abstraction layer for the ARM Cortex®-M processor series and defines
+generic tool interfaces. Though it was originally defined for ARM Cortex-M
+microcontrollers alone, it could be easily extended to other microcontrollers
+making it generic. For more information on CMSIS RTOS v2, please refer
+http://www.keil.com/pack/doc/CMSIS/RTOS2/html/index.html
+
+Features not supported in Zephyr implementation:
```

Comment:
Needs some formatting fixes (I've laid this out as a "dictionary list", where the first line will be bold and the subsequent indented lines will be an indented paragraph):

    Features not supported in Zephyr implementation
    ***********************************************

    Kernel
       osKernelGetState, osKernelSuspend, osKernelResume are not supported.
       Also, osKernelInitialize and osKernelStart are pretty much empty.

    Threads
       osThreadJoin and osThreadDetach are not supported.

    Mutex
       osMutexPrioInherit is supported by default and is not configurable,
       we cannot select/unselect this attribute.
       osMutexRecursive is also supported by default. If this attribute is
       not set, an error is thrown when the same thread tries to acquire
       it the second time.
       osMutexRobust is not supported in Zephyr.

    Return values not supported in the Zephyr implementation
    ********************************************************

    osSemaphoreDelete
       osErrorResource (the semaphore specified by parameter
       semaphore_id is in an invalid semaphore state) is not supported.

    osMutexDelete
       osErrorResource (mutex specified by parameter mutex_id
       is in an invalid mutex state) is not supported.

    osTimerDelete
       osErrorResource (the timer specified by parameter timer_id
       is in an invalid timer state) is not supported.

    osMessageQueueReset
       osErrorResource (the message queue specified by
       parameter msgq_id is in an invalid message queue state)
       is not supported.

    osMessageQueueDelete
       osErrorResource (the message queue specified by
       parameter msgq_id is in an invalid message queue state)
       is not supported.

    osMemoryPoolFree
       osErrorResource (the memory pool specified by
       parameter mp_id is in an invalid memory pool state) is
       not supported.

    osMemoryPoolDelete
       osErrorResource (the memory pool specified by
       parameter mp_id is in an invalid memory pool state) is
       not supported.

    osEventFlagsSet, osEventFlagsClear
       osFlagsErrorUnknown (Unspecified error)
       and osFlagsErrorResource (Event flags object specified by
       parameter ef_id is not ready to be used) are not supported.

    osEventFlagsDelete
       osErrorParameter (the value of the parameter ef_id is
       incorrect) is not supported.

    osThreadFlagsSet
       osFlagsErrorUnknown (Unspecified error) and
       osFlagsErrorResource (Thread specified by parameter
       thread_id is not active to receive flags) are not supported.

    osThreadFlagsClear
       osFlagsErrorResource (Running thread is not active to
       receive flags) is not supported.

Reply:
Done.